### PR TITLE
chore(ICP-Rosetta): FI-1559: icp rosetta client add and remove hotkey

### PR DESCRIPTION
**Pull request by NikolasHai**
This MR proposes the following changes:

1. Add a functionality to add and remove hotkeys for a neuron to ICP Rosetta client
2. Migrate the legacy test for hotkey management to pocket ic

### CODE REVIEW

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 1]
```diff
@@ -30,6 +35,7 @@ use tokio::runtime::Runtime;
 
 lazy_static! {
     pub static ref TEST_IDENTITY: Arc<BasicIdentity> = Arc::new(test_identity());
+    pub static ref NUM_TEST_CASES: u32 = 1;
```
Comment (mbjorkqvist) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:38]: nit: Could this be a plain `const` outside of the `lazy_static`?

File: rs/tests/financial_integrations/rosetta/BUILD.bazel [hunk 2]
```diff
@@ -40,27 +40,6 @@ system_test_nns(
...
 )
 
-system_test_nns(
-    name = "rosetta_neuron_hotkey_test",
-    extra_head_nns_tags = [],  # don't run the head_nns variant on nightly since it aleady runs on long_test.
-    flaky = True,
-    proc_macro_deps = MACRO_DEPENDENCIES,
-    tags = [
-        "k8s",
-        "long_test",  # since it takes longer than 5 minutes.
```
Comment (mbjorkqvist) [inline rs/tests/financial_integrations/rosetta/BUILD.bazel:50]: Very nice that we can get rid of this (for all the migrated tests)!

**Review by mbjorkqvist — APPROVED** [review]
Thanks @NikolasHai !

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs (hunk 1 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:38]: Yes, but is there a benefit to that?

Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:38]: Okay, thanks for pointing that out. I can create a ticket to change the pattern for all tests.

**Review by marko-k0 — APPROVED** [review]

File: rs/rosetta-api/icp/client/src/lib.rs [hunk 3]
```diff
@@ -397,6 +398,74 @@ impl RosettaClient {
...
+                        _ => bail!("Either public key or principal id has to be set"),
+                    },
+                }
+                .try_into()
+                .map_err(|e| anyhow::anyhow!("Failed to convert metadata: {:?}", e))?,
+            ),
+        }])
+    }
+
+    pub fn build_remove_hot_key_operations(
```
Comment (fsodre) [inline rs/rosetta-api/icp/client/src/lib.rs:435]: It seems to me this function is exactly the same as `build_add_hot_key_operations` except for the `type_` field. Could we factor this out into a separate generic function?

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 4]
```diff
@@ -816,3 +822,154 @@ fn test_get_neuron_info() {
...
+
+                    let neuron = list_neurons(&agent)
+                        .await
+                        .full_neurons
+                        .first()
+                        .unwrap()
+                        .to_owned();
+                    // Make sure that the neuron has no hot keys
+                    assert!(neuron.hot_keys.is_empty());
+                });
```
Comment (fsodre) [inline rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs:970]: Let's also test the behavior for invalid inputs: no key provided and both keys provided.

File: rs/rosetta-api/icp/client/src/lib.rs (hunk 3 above)
Comment (NikolasHai) [reply rs/rosetta-api/icp/client/src/lib.rs:437]: We could, however I designed these build functions specifically for the purposes of readability for external users. The plan is to release the ICP Rosetta client to crates.io and the build operations are essential for the understanding of how transactions with ICP Rosetta are built. extracting logic here adds complexity and reduces readability while shortening the number of lines of codes slightly. I do not think that the benefit outway the cons here.

**Pull request merged**